### DISCUSSION TITLE
add max tags or checkbox limit request

### DISCUSSION THREAD

**Discussion by Thedineshk24** [dowjones/react-dropdown-tree-select#571] [Q&A]
there should be a prop called maxTagsLimit OR maxCheckbox limit should be available , currently there are no prop available in the library. so I was not able to restrict user to stop selecting checkbox.

EX: suppose I want user to select only 6 checkbox and after that disable other checkboxes but currently I'm not able to do it.

**Comment by mrchief** [comment]
You can do it via an HoC. Something like this would work:

 - keep track of number of selections
 - once the set limit is reached, the onChange handler becomes a no-op

**Comment by Thedineshk24** [comment]
can u please share some sample code for that, because I've tried to stop
adding more data in onChange after limit is reached, but I'm still able to
select more checkboxes

**Comment by mrchief** [comment]
I can imagine something like this

```
maxCheckable = 5
maxChecked = 0

onChange {
  if maxChecked > maxCheckable {
    node.selected = false
    // update data
    // use tree node paths HoC for keeping track of state on your end
   setData(updatedData)
   return 
  }
  if node.selected == true; maxChecked++
  // update data but no need to do setData
}
```

**Reply by Thedineshk24** [reply]
> node

Hi, I've tried your solution but it's not working. here is codesandboxlink 
[https://codesandbox.io/s/using-bootstrap-styles-with-react-dropdown-tree-select-forked-tmsw60?file=/index.js](https://codesandbox.io/s/using-bootstrap-styles-with-react-dropdown-tree-select-forked-tmsw60?file=/index.js)

**Comment by RivkindLeon** [comment]
Any way how I can limit showed tags? Something like to show first 2 tags, and for next just note how much( example:  AUTO, BUS, +3) or at least show dots(...). If I have lots of selected tags my input components change size and this behavior not expected for me

**Reply by mrchief** [reply]
It should be doable with a bit of css - you can target `:nth-child(n+2)` nodes and hide them.
